app/internal/wm/os_js.go

```diff
@@ -524,6 +545,12 @@ func (w *window) ShowTextInput(show bool) {
 	}()
 }
 
+func (w *window) SetInputHint(mode key.InputHint) {
+	go func() {
+		w.keyboard(mode)
+	}()
```

Comment:
SetInputHint should be called from in Window.Run and you don't need a goroutine here anymore.